### PR TITLE
fp: Dispute Game Blacklist task

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> [!NOTE]
> Stacked PR:
> #185
> #184
> #183 👈
> `main`

Adds a task on Sepolia for blacklisting dispute games from the `Guardian` safe.

### CODE REVIEW

**Review by Inphi — DISMISSED** [review]

File: tasks/sep/fp-recovery/001-blacklist-game/SignFromJson.s.sol [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        string memory inputJson;
+        string memory path = "/tasks/sep/fp-recovery/001-blacklist-game/input.json";
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
+            inputJson = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+
+        address blacklistedGame = stdJson.readAddress(inputJson, "$.transactions[0].contractInputsValues._disputeGameProxy");
+        require(!portal.disputeGameBlacklist(IDisputeGame(blacklistedGame)), "Dispute game is not blacklisted");
```
Comment (Inphi) [inline tasks/sep/fp-recovery/001-blacklist-game/SignFromJson.s.sol:52]: Shouldn't we expect the game to be blacklisted after executing this runbook? Looks like the condition is flipped.

Comment (clabby) [reply tasks/sep/fp-recovery/001-blacklist-game/SignFromJson.s.sol:52]: Ah whoops - committed after checking the inverse to see that it failed properly 🤦‍♂️ 

Will update when back at desk.

File: script/DeployRehearsalContracts.s.sol [hunk 2]
```diff
@@ -33,7 +33,7 @@ contract DeployRehearsalContracts is Deployer {
         require(owner_safe.isOwner(address(council_safe)));
 
         console.log("Deploying from %s", name());
-        console.log("Deployment context: %s", Config.deploymentContext());
+        console.log("Deployment context: %s", vm.envOr("DEPLOYMENT_CONTEXT", string("")));
```
[the remark this answers is not in the document]
Comment (clabby) [reply script/DeployRehearsalContracts.s.sol:36]: This is intentional - they removed this helper in the monorepo

**Review by Inphi — APPROVED** [review]

File: tasks/sep/fp-recovery/001-blacklist-game/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+The batch will be executed on chain ID `11155111`, and contains `1` transactions.
+
+### Tx #1: Blacklist Dispute Game
+
+Executes the `blacklistDisputeGame` call to the `OptimismPortal` proxy
+
+**Function Signature:** `blacklistDisputeGame(address)`
+
+**To:** `0x16Fc5058F25648194471939df75CF27A2fdC48BC`
```
[the remark this answers is not in the document]
Comment (clabby) [reply tasks/sep/fp-recovery/001-blacklist-game/README.md:15]: Ack - going to merge to unblock some work, need the changes to support the latest contracts release. Will follow up.

File: tasks/sep/fp-recovery/001-blacklist-game/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0x7d6be8dc000000000000000000000000deaddeaddeaddeaddeaddeaddeaddeaddeaddead`
```
[the remark this answers is not in the document]
Comment (clabby) [reply tasks/sep/fp-recovery/001-blacklist-game/README.md:19]: https://github.com/ethereum-optimism/superchain-ops/pull/183#discussion_r1600829844

**Review by Inphi — DISMISSED** [review]

File: tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+The batch will be executed on chain ID `11155111`, and contains `1` transactions.
+
+## Tx #1: Update `respectedGameType` in the `OptimismPortal`
+
+Updates the `respectedGameType` to `PERMISSIONED_CANNON` in the `OptimismPortal`, enabling permissioned proposals and challenging.
+
+**Function Signature:** `setRespectedGameType(address,uint32)`
+
+**To:** `0x4220C5deD9dC2C8a8366e684B098094790C72d3c`
```
Comment (Inphi) [inline tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md:13]: This is the `DeputyGuardianModule`. It should be the portal at `0x16Fc5058F25648194471939df75CF27A2fdC48BC`,

Comment (clabby) [reply tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md:13]: Spoke offline; Should be the guardian module

Comment (Inphi) [reply tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md:13]: nvmind. I see we're going through the module from now on. good looks.

**Review by Inphi — APPROVED** [review]

**Comment by Inphi** [comment]
@maurelian Should be good for another review.

File: tasks/sep/fp-recovery/001-blacklist-game/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+1. Generate the batch with `just generate-input <address>`.
+
+1. Collect signatures for the action.
+
+1. Execute the action.
+
+## Signing and execution
+
+Please see the signing and execution instructions in [SINGLE.md](../../../../SINGLE.md).
```
Comment (maurelian) [inline tasks/sep/fp-recovery/001-blacklist-game/README.md:39]:
```suggestion
1. Collect signatures and execute the action according to the instructions in [SINGLE.md](../../../../SINGLE.md).
```

File: tasks/sep/fp-recovery/001-blacklist-game/out.md
Comment (maurelian) [inline tasks/sep/fp-recovery/001-blacklist-game/out.md:1]: It looks like the content of this file is identical to what's in the README, can we remove it please?

File: tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+## Preparing the Operation
+
+1. Collect signatures for the action.
+
+1. Execute the action.
+
+## Signing and execution
+
+Please see the signing and execution instructions in [SINGLE.md](../../../../SINGLE.md).
```
Comment (maurelian) [inline tasks/sep/fp-recovery/002-fallback-permissioned-game/README.md:33]: Similar to the prev comment, these two sections are redundant. Let's clean it up please.

File: tasks/sep/fp-recovery/003-enable-permissionless-game/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+## Preparing the Operation
+
+1. Collect signatures for the action.
+
+1. Execute the action.
+
+## Signing and execution
+
+Please see the signing and execution instructions in [SINGLE.md](../../../../SINGLE.md).
```
Comment (maurelian) [inline tasks/sep/fp-recovery/003-enable-permissionless-game/README.md:33]: Same

File: tasks/sep/fp-recovery/003-enable-permissionless-game/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+### State Validations
+
+The only state modification that is made by this action is an update to the shared slot between the `respectedGameType`
+and `respectedGameTypeUpdatedAt` variables:
+
+![state-diff](./images/state_diff.png)
+
+Slot `0x000000000000000000000000000000000000000000000000000000000000003b` in the `OptimismPortal` proxy has the following packed layout:
```
Comment (maurelian) [inline tasks/sep/fp-recovery/003-enable-permissionless-game/README.md:42]: In each of the 3 readme files, please link to the location in the storage layout lockfiles which will attest to the relevant slots.

**Review by maurelian — COMMENTED** [review]
Looks good, a few comments to polish it up.

**Review by maurelian — APPROVED** [review]

**Pull request merged**